lib/node/cli/args/args-setter.ts

```diff
@@ -0,0 +1,38 @@
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import { CliArgumentsSetter, Command, CommandOption } from '../cli.models.js';
+
+export function argumentsSetter(): CliArgumentsSetter {
+    const argv = yargs(hideBin(process.argv));
+
+    return {
+        withCommand(command: Command): CliArgumentsSetter {
```

Comment:
I feel like the API I used is much cleaner and nicer then yargs. In general I try to abstract from 3rd party libs when I can because it makes it easier to update / replace the library in future. This might not be a huge concern here, but the API itself makes it worth the effort to me :)

Reply:
I'll have to disagree here a bit because it actually makes a lot of sense - this comes from my experience of working on a few larger projects (outside K.ai) where one of the best decisions we ever made was to create simple abstractions for most 3rd party packages we rely on. In JS ecosystem things tend to change quite quickly and often and it might be difficult to keep everything up-to-date. 

By creating these simple abstractions (even if they are very simple "wrappers") you can centralize all the calls to 3rd party libs as well as just exposing only what you need. As a bonus, you can create a tad nicer API as well. 

If something changes, you don't need to go around the entire solution to "fix" everything, you just fix it in a single, centralized, location. Saves a lot of time and effort. Especially if there are "changes" you might need to make along the way. It also makes switching between libraries very easy as well. If something new and shiny comes out, it gives you the ability to easily try it, or switch to it. 

For this specific use case here, sure, it's not necessary to have this wrapper. I don't expect to change yargs (though one never knows :D ) and it's a simple enough API that I could just use it as it is. Thing is, I don't like yarg's API and wanted to spend the extra hour or so by creating a bit nicer API for me and my purposes.

Reply:
Sure, I don't think that creating facades around packages is bad in general I just think that this one is an overhead rather than an improvement.

When doing anything, you have to consider the specifics of the project/situation and act accordingly. Yargs is a CLI library and as such, it is isolated from the rest of the project even without any facade around it.

What do you dislike about the yarg's API that this wrapper is worth it? I mean, the wrapper is almost identical to the original yarg's API. That's another reason why the wrapper is of little use. If the yarg's API changes, you have a wrapper following the old API. Such wrappers should follow domain specifics instead of the wrapped library design.

In my opinion, it increases the cognitive load of the project without bringing value.

Reply:
It's not as isolated, it's still part of the solution, it doesn't really matter if you categorize it as CLI or not. To me, its an improvement over the yargs API in its simplicity/extensibility/readability.  You might disagree, and that's okay, you didn't write the API so it's foreign to you. 

I could also add a few more types here and there to increase type safety for specific command/option mappings if I wanted to.

I think that if it's a simple enough wrapper that takes you an hour to write, the benefits will be worth it. As always, things are subjective and people can have different views on things, I'm not saying you are wrong or anything, it's just that I, personally, like this approach more :)